codec-http2/src/main/java/io/netty/handler/codec/http2/HpackStaticTable.java

```diff
@@ -147,20 +149,26 @@ static int getIndex(CharSequence name) {
      */
     static int getIndexInsensitive(CharSequence name, CharSequence value) {
         int index = getIndex(name);
-        if (index == -1) {
-            return -1;
+        if (index == NOT_FOUND) {
+            return NOT_FOUND;
         }
 
         // Note this assumes all entries for a given header field are sequential.
         while (index <= length) {
             HpackHeaderField entry = getEntry(index);
-            if (equalsVariableTime(name, entry.name) && equalsVariableTime(value, entry.value)) {
+            if (!equalsVariableTime(name, entry.name)) {
+                // As far as fields with the same name are placed in the table sequentialy
+                // and INDEX_BY_NAME returns index of the fist position, - it's safe to
+                // exit immediatly.
+                return NOT_FOUND;
+            }
+            if (equalsVariableTime(value, entry.value)) {
                 return index;
             }
             index++;
```

Comment:
Optional idea: at index values above 15 have no other duplicate values.  I think it should be possible to break the loop here if the index is above 15